### PR TITLE
Rewrite the restaurant plugin in Python

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
With the Refter changing its operations, and the UMC's menu now being published on their website, I wanted to rewrite the restaurant plug-in in a more modular fashion.

Currently, the following restaurants are supported:

* Het Gerecht (Grotius building)
* Radboud UMC hospital restaurant
* Cafetaria De Fest

Support for De Refter will be re-introduced when it reopens. Its new 'food court' nature is
likely to have consequences for the presentation of its menu, so it has been left unimplemented for now.

Work in progress — it just loads the menus for now. Adding the bot interactions shouldn't be too much trouble, though.

### CODE REVIEW

File: suppliers/hetgerecht.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        # Fridays from 11:00 until 16:00
+        elif dt.weekday() == 4:
+            return dt.hour >= 11 and dt.hour < 16
+
+        # ... What planet is this?
+        else:
+            raise ValueError('Encountered an unexpected weekday.')
+
+    @staticmethod
+    def willOpen(dt=datetime.now()):
```
Comment (sgielen) [inline suppliers/hetgerecht.py:67]: If dt.weekday() == 5?

File: suppliers/hetgerecht.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        elif dt.weekday() < 4:
+            return dt.hour >= 11 and dt.hour < 19
+
+        # Fridays from 11:00 until 16:00
+        elif dt.weekday() == 4:
+            return dt.hour >= 11 and dt.hour < 16
+
+        # ... What planet is this?
+        else:
+            raise ValueError('Encountered an unexpected weekday.')
```
Comment (sgielen) [inline suppliers/hetgerecht.py]: This isn't really necessary, but it's up to you :P

**Review by sgielen — COMMENTED** [review]
Looks nice, just two small comments. And I'd like some testing on IRC if you can still run this somewhere :)

File: dazeus-noms.py [hunk 3]
```diff
@@ -139,57 +192,159 @@ def getMenu(dt=datetime.now()):
...
+
+        # Ensure it is one we know.
+        supplier = params[5].lower()
+        if supplier not in suppliers:
+            reply('I don\'t have a menu for that place... How about one of these: ' + ', '.join(suppliers))
+            return
+
+        # Did we say when we'd like to eat?
+        now = datetime.now();
+        if len(params) >= 7:
```
Comment (sgielen) [inline dazeus-noms.py:42]: You already tested for len(params) < 6 above, so this will always be true here.

File: suppliers/hetgerecht.py (hunk 1 above)
Comment (AaronVanGeffen) [reply suppliers/hetgerecht.py:67]: Ah, de eerste regel moet `<=` zijn idd. Nice catch.

File: suppliers/hetgerecht.py (hunk 2 above)
Comment (AaronVanGeffen) [reply suppliers/hetgerecht.py]: Yeah, noticed just now that it's unreachable in this code path. Will remove.

File: dazeus-noms.py (hunk 3 above)
Comment (AaronVanGeffen) [reply dazeus-noms.py:42]: This won't be true if len(params) == 6, when only a supplier name is provided, right? :)

Comment (sgielen) [reply dazeus-noms.py:42]: Oh, you're right, sorry!

**Pull request merged**